### ISSUE TITLE
Wrong Typescript type for Query with a variable with default value

### ISSUE THREAD

**Issue by jflewis** [dotansimha/graphql-code-generator#1934]
**Describe the bug**
 
When generating Typescript types for a query with default variables, the generated query variables interface specifies the field as `Maybe<Scalars["Boolean"]>`. However, I believe the correct implementation should be `field?: Scalars["Boolean']` so the field isn't required when defining the variables in a query. Having the field be mandatory for a optional field doesn't seem quite right.

**To Reproduce**

1. My GraphQL schema:

```graphql
# Put your schema here
type Query {
    kit(id: ID!): Kit!
  }
```

2. My GraphQL operations:

```graphql
# Put your operations here
query kit($withRawData: Boolean = false, $id: ID!, $languageId: ID!) {
    kit(id: $id) {
      id
      kitId
      nickname
      status
      trackingNumber
      deliveryService
      personalizedReports {
        id
        reportType
      }
      data(languageId: $languageId)
      rawData @include(if: $withRawData) {
        id
        rawDataType
        url
      }
    }
  }
```

3. My `codegen.yml` config file:

```yml
# Put your YML here
pluckConfig:
  modules:
    - name: "apollo-boost"
      identifier: "gql"
schema: http://localhost:3001
documents: graphql/**/*.{ts,tsx}
generates:
  generated/apolloComponents.tsx:
    plugins:
      - typescript
      - typescript-operations
      - typescript-react-apollo
```

**Expected behavior**

**Environment:**

- OS: mac os
- `@graphql-codegen/...`:  "^1.0.6"
- NodeJS: 8.10

**Additional context**

example.) compiler fails with missing property of `withRawData`
```
<KitComponent
              variables={{
                id: kit.id,
                languageId: user.language.id || DEFAULT_LANGUAGE_ID
              }}
```

**Comment by Aleksion** [comment]
Seeing the same thing

**Comment by n1ru4l** [comment]
@Aleksion @jflewis I tackled this issue in https://github.com/dotansimha/graphql-code-generator/pull/2018 

You could help me out by verifying it is behaving correct in a real-world application.

**Comment by dotansimha** [comment]
Fixed by @n1ru4l ❤️ 
If someone wants to test it before we release a stable version, you can use `1.2.2-alpha-d6b07f60.28`.

**Comment by dotansimha** [comment]
Fixed in 1.3.0 🎉
